src/components/ProductTour/ProductTour.tsx

```diff
@@ -0,0 +1,155 @@
+import * as React from 'react';
+import { bem } from '../../utils';
+import { ENTER_KEY } from '../../constants';
+import { Button } from '../Buttons';
+import { Checkbox } from '../Checkbox';
+import { Modal, ModalProps } from '../Modal';
+import styles from './ProductTour.scss';
+
+interface Props extends ModalProps {
+    /** Should the tour be shown */
+    isOpen: boolean;
+    /** An array of ReactElement where each element is one slide */
+    children: React.ReactElement | React.ReactElement[];
+    /** The label for the checkbox. If undefined or empty, checkbox will not be shown */
+    checkboxLabel?: string;
+    /** Cancel button label - required if there is more then 1 slide */
+    cancelLabel?: string;
+    /** Continue button label - required if there is more then 1 slide */
+    continueLabel?: string;
+    /** Finish button label */
+    finishLabel: string;
+    /** A callback if the user clicked the Cancel button without finishing the tour */
+    onCancel?: (isChecked?: boolean) => void;
+    /** A callback if the user clicked the Finish button */
+    onFinished: (isChecked?: boolean) => void;
+}
+
+const { block, elem } = bem('ProductTour', styles);
+
+export const ProductTour: React.FC<Props> = (props) => {
+    const {
+        isOpen,
+        children,
+        checkboxLabel,
+        cancelLabel,
+        continueLabel,
+        finishLabel,
+        onCancel,
+        onFinished,
+        ...rest
+    } = props;
+
+    /** index for currently shown slide; 0 indexed */
+    const [currentSlide, setCurrentSlide] = React.useState(0);
+    /** state of the checkbox */
+    const [isChecked, setIsChecked] = React.useState(false);
+    const nOfSlides = React.Children.count(children);
+
+    if (nOfSlides > 1 && (!cancelLabel || !continueLabel)) {
+        // eslint-disable-next-line no-console
+        console.warn(
+            'ProductTour component was called with incomplete props. When there are 2 or more slides in the tour, you need to include both `cancelLabel` and `continueLabel` as props'
+        );
+    }
+
+    const isLastSlide = currentSlide === nOfSlides - 1;
+
+    const handleCheckboxClick = () => {
+        setIsChecked(!isChecked);
+    };
+
+    const handleDotClick = (i: number) => {
+        setCurrentSlide(i);
+    };
+
+    const handleDotKeyDown = (e: React.KeyboardEvent<HTMLElement>, i: number) => {
+        if (e.key === ENTER_KEY) {
+            handleDotClick(i);
+        }
+    };
+
+    const handleCancel = () => {
+        onCancel?.(isChecked);
+    };
+
+    const handleNext = () => {
+        setCurrentSlide(currentSlide + 1);
+    };
+
+    const handleFinish = () => {
+        onFinished(isChecked);
```

Comment:
Very much of a corner case but still FYI:
Consider you have checked the checkbox, then prop `checkboxLabel` is changed to `undefined`, `isChecked` will still be `true` while there is no checkbox.

Reply:
Indeed an edge case. I can add code to not return anything if there is no checkbox. For now I assumed the application knows if they want to read this variable or not. It might make it easier for TS if the def always have the (last) value of the checkbox sent over. But if you think there is real danger here, I can remove it.

Reply:
No danger :) just so you know